crates/bevy_ui/src/focus.rs

```diff
@@ -155,7 +157,7 @@ pub fn ui_focus_system(
                 // clicking
                 let contains_cursor = if let Some(cursor_position) = cursor_position {
                     (min.x..max.x).contains(&cursor_position.x)
-                        && (min.y..max.y).contains(&cursor_position.y)
+                        && (min.y..max.y).contains(&(window_height - cursor_position.y))
```

Comment:
I think everything that works with 2D still has the old orientation, so cursor position would still be bottom-left based for them.  Is that asssumption valid?

Reply:
I think it makes sense for the "cursor position" and "touch position" to stay bottom-to-top, as those input events are used by people for implementing their 2D game mechanics and such, and the main camera axis has +Y going up.

Hence, I think this PR is correct in performing an extra inversion for UI only.

If we change the cursor events, that will affect not just UI, but also 2d gameplay code.

Reply:
True, but world space in 2D will only match that of the cursor if the camera is not translated, rotated, scaled, or has a non-default scaling mode, so people would need to convert from camera to world space anyway.

For the UI though as long as we don't invert the y co-ordinate it matches. This also simplifies the internals a bit and prevents third party-crates from needing to invert the y co-ordinate also.

Reply:
If I understand correctly we have two use cases:
2D cameras and this specific UI projection matrix.

2D camera: 
When I place a sprite at (50,50) it will be placed in the lower left corner of the view.
Everyone who deals with mouse coordinates here might be surprised that the mouse coordinates do not match with where they place a sprite.

UI:
This is more the exception than the rule. I would consider handling mouse coordinates with UI an advanced usecase, where you can more reasonably expect a user invert the Y axis.

The reason why I would rather leave the flipping of the coordinate in the UI system is:
- I think UI with mouse events is a less common usecase.
- If it is preferable to also invert the mouse coordinates I would rather do that in a separate PR, so this PR stays tiny

Reply:
@devil-ira Alright! I think your suggestion about inverting cursor works like it should. 
I'll fix the examples now and update the screenshots in the description. Anything else I should consider?